script.js

```diff
@@ -0,0 +1,245 @@
+      updateDisplay(calcSubtraction(savedInputNumber, currentNumber));
+      break;
+    case "multiplication":
+      updateDisplay(calcMultiplication(savedInputNumber, currentNumber));
+      break;
+    case "division":
+      updateDisplay(calcDivision(savedInputNumber, currentNumber));
+      break;
+  }
+  resultNumber = inputNumberField.innerText;
+  savedInputNumber = resultNumber;
+  replaceNumber = true;
+  currentNumber = 0
+  removeActiveStyle();
+};
+
+const updateDisplay = (input) => {
+  if (input.toString().length >= 16) {
+    inputNumberField.innerText = parseFloat(input).toFixed(14);
+  } else {
+    inputNumberField.innerText = input;
+  }
+};
+
+const setOperator = (input) => {
+  calculate();
+  replaceNumber = true;
+  savedInputNumber = inputNumberField.innerText;
+  inputOperator = input;
+  setActiveStyle(input);
+};
+
+const setActiveStyle = (input) => {
+  removeActiveStyle();
+  const styleButton = document.querySelector(`.${input}`);
+  styleButton.classList.add("active");
+};
+
+const removeActiveStyle = () => {
+  operationButtons.forEach((operationButton) => {
+    operationButton.classList.remove("active");
+  });
+};
+
+const changeSign = () => {
+  if (Math.sign(currentNumber) == 1) {
+    currentNumber = Math.abs(currentNumber) * -1;
+  } else {
+    currentNumber = Math.abs(currentNumber) * 1;
+  }
+  inputNumberField.innerText = currentNumber;
+};
+
+const appendInput = (input) => {
+  if (inputNumberField.innerText.length > 14) {
+    return;
+  }
+  inputNumberField.innerText += input;
+  currentNumber = inputNumberField.innerText;
+};
+
+const replaceInput = (input) => {
+  inputNumberField.innerText = input;
+  currentNumber = inputNumberField.innerText;
+};
+
+const calcAddition = (num1, num2) => {
+  return parseFloat(num1) + parseFloat(num2);
+};
+
+const calcSubtraction = (num1, num2) => {
+  return parseFloat(num1) - parseFloat(num2);
+};
+
+const calcMultiplication = (num1, num2) => {
+  return parseFloat(num1) * parseFloat(num2);
+};
+
+const calcDivision = (num1, num2) => {
+  return parseFloat(num1) / parseFloat(num2);
+};
+
+const deleteNumber = () => {
+  inputNumberField.innerText = inputNumberField.innerText.slice(0, -1);
+  if (inputNumberField.innerText.length == 0) {
+    replaceNumber = true;
+    inputNumberField.innerText = 0;
+  } else if (inputNumberField.innerText == "0") {
+    replaceNumber = true;
+  }
+  currentNumber = inputNumberField.innerText;
+};
+
+const clearAll = () => {
+  removeActiveStyle();
+  inputNumberField.innerText = 0;
+  replaceNumber = true;
+  inputOperator = undefined;
+  savedInputNumber = undefined;
+  currentNumber = undefined;
+  resultNumber = undefined;
+};
+
+const isDigit = (input) => {
+  return /\d/.test(input);
+};
+
+const keyPressEvent = (input) => {
+  const number = document.querySelector(`[data-number="${input}"]`);
+  number.classList.add("keydown");
+};
+
+numberButtons.forEach((button) => {
+  button.ontransitionend = (e) => {
```

Comment:
This is not reliable, it will get stuck in the keydown state if you press a key for a longer time.

Reply:
Fixed in latest commit